### PR TITLE
fix(hacbs-1522): update enterprise contract dependency

### PR DESCRIPTION
**Pull request by scoheb**
Attempting to fix:

E1219 20:22:51.699193 1 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1alpha1.EnterpriseContractPolicy: failed to list *v1alpha1.EnterpriseContractPolicy: json: cannot unmarshal object into Go struct field EnterpriseContractPolicySpec.items.spec.sources of type string

### CODE REVIEW

File: controllers/release/release_adapter_test.go [hunk 1]
```diff
@@ -70,7 +70,14 @@ var _ = Describe("Release Adapter", Ordered, func() {
 			},
 			Spec: ecapiv1alpha1.EnterpriseContractPolicySpec{
 				Description: "test-policy-description",
-				Sources:     []string{"foo"},
+				Sources: []ecapiv1alpha1.PolicySource{
```
Comment (johnbieren) [inline controllers/release/release_adapter_test.go:73]: This should just be `Source` not `PolicySource` based on how they have things now

File: controllers/release/release_adapter_test.go [hunk 2]
```diff
@@ -70,7 +70,14 @@ var _ = Describe("Release Adapter", Ordered, func() {
...
+					ecapiv1alpha1.PolicySource{
```
Comment (johnbieren) [inline controllers/release/release_adapter_test.go:74]: Source not PolicySource

Comment (johnbieren) [reply controllers/release/release_adapter_test.go:74]: Inside of the struct, you don't want a GitRepository like this, you want (at most)
```
Name: string
Policy: 
- string
Data: 
- string
```

File: tekton/pipeline_run_test.go [hunk 3]
```diff
@@ -105,7 +105,14 @@ var _ = Describe("PipelineRun", func() {
 			},
 			Spec: ecapiv1alpha1.EnterpriseContractPolicySpec{
 				Description: "test-policy-description",
-				Sources:     []string{"foo"},
+				Sources: []ecapiv1alpha1.PolicySource{
```
Comment (johnbieren) [inline tekton/pipeline_run_test.go:108]: same comments as other test

**Review by johnbieren — CHANGES_REQUESTED** [review]

File: tekton/pipeline_run_test.go [hunk 4]
```diff
@@ -105,7 +105,12 @@ var _ = Describe("PipelineRun", func() {
 			},
 			Spec: ecapiv1alpha1.EnterpriseContractPolicySpec{
 				Description: "test-policy-description",
-				Sources:     []string{"foo"},
+				Sources: []ecapiv1alpha1.Source{
+					ecapiv1alpha1.Source{
+						Name:   "foo",
+						Policy: []string{"https://github.com"},
```
Comment (simonbaird) [inline tekton/pipeline_run_test.go:111]: I assume the actual value doesn't matter, but for readability, aybe a more realistic is better:

    Policy: []string{"https://github.com/company/policy"},

Comment (simonbaird) [reply tekton/pipeline_run_test.go:111]: There's also a "Data" key:

    Data: []string{"https://github.com/some-company/data"},

File: controllers/release/release_adapter_test.go (hunk 2 above)
Comment (scoheb) [reply controllers/release/release_adapter_test.go]: Fixed

File: tekton/pipeline_run_test.go (hunk 3 above)
Comment (scoheb) [reply tekton/pipeline_run_test.go]: Fixed

File: tekton/pipeline_run_test.go (hunk 4 above)
Comment (simonbaird) [reply tekton/pipeline_run_test.go]: Yep, that's fine. (I mentioned just as an fyi.)

File: go.sum [hunk 5]
```diff
@@ -218,8 +218,8 @@ github.com/googleapis/gax-go/v2 v2.0.5/go.mod h1:DWXyrwAJ9X0FpwwEdw+IPEYBICEFu5m
 github.com/grpc-ecosystem/grpc-gateway v1.14.6/go.mod h1:zdiPV4Yse/1gnckTHtghG4GkDEdKCRJduHpTxT3/jcw=
 github.com/grpc-ecosystem/grpc-gateway v1.16.0 h1:gmcG1KaJ57LophUzW0Hy8NmPhnMZb4M0+kPpLofRdBo=
 github.com/grpc-ecosystem/grpc-gateway v1.16.0/go.mod h1:BDjrQk3hbvj6Nolgz8mAMFbcEtjT1g+wF4CSlocrBnw=
-github.com/hacbs-contract/enterprise-contract-controller/api v0.0.0-20221123200519-210c9331b2e2 h1:dIjGMlLHfFsGk6FqLpv7foxGU113bjZW3cXrurFMsjw=
-github.com/hacbs-contract/enterprise-contract-controller/api v0.0.0-20221123200519-210c9331b2e2/go.mod h1:KIVowf1Q+rkLyFdcquUHT/4QIov9j5C4TgGJ/jgT3io=
+github.com/hacbs-contract/enterprise-contract-controller/api v0.0.0-20221215171221-93099d5a1eff h1:UnXu7RA6twiM8nXbx/LZH/vc41RKNUvnPRFCmCqIv7Q=
+github.com/hacbs-contract/enterprise-contract-controller/api v0.0.0-20221215171221-93099d5a1eff/go.mod h1:KIVowf1Q+rkLyFdcquUHT/4QIov9j5C4TgGJ/jgT3io=
```
Comment (simonbaird) [inline go.sum:222]: I didn't check the digests, but beware there was a fix merged just a few seconds ago. Maybe it needs to refresh.

Comment (simonbaird) [reply go.sum:222]: The fix included updated CRDs, see https://github.com/hacbs-contract/enterprise-contract-controller/pull/48

File: tekton/pipeline_run_test.go (hunk 4 above)
Comment (scoheb) [reply tekton/pipeline_run_test.go]: thanks. added Data

**Comment by johnbieren** [comment]
As discussed in slack, it won't pass CI due to the controller-runtime version mismatch, but you should run `go get github.com/hacbs-contract/enterprise-contract-controller/api`. I expect the `go.mod` diff to be larger

**Review by johnbieren — APPROVED** [review]

**Pull request merged**